### PR TITLE
[Java][Server] fix default value

### PR DESCRIPTION
**Pull request by borsch**
Introduce new `DefaultCode.toDefaultParameterValue(Schema)`. This method will be used to generate default value for method arguments. Default implementation use old `toDefaultValue(Schema)` to be backward compatible
Fix for: https://github.com/OpenAPITools/openapi-generator/issues/8535

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by borsch** [comment]
@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala  (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01)  @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @nmuesch  (2021/01)

**Comment by Zomzog** [comment]
I'm not sure to understand why you add a new property, why not fixing directly the default value?
Maybe with a method that can be overridden by other generators like it's done for examples.

**Comment by borsch** [comment]
@Zomzog you can see here https://github.com/OpenAPITools/openapi-generator/pull/8577/files#diff-df8051ffa04f7a4baff1a9795bf531584fa9b4bac81c2667a3c0f5ee44172dbcR181

`defaultValue` doesn't contains original literal from yaml/json schema but rather default object for specific language

**Comment by wing328** [comment]
What about overriding the `toDefaultValue` method to accomplish what you need?

ref: https://github.com/OpenAPITools/openapi-generator/search?q=toDefaultValue

**Comment by borsch** [comment]
@wing328 @Zomzog I have changed implementation

**Review by Zomzog — APPROVED** [review]
LGTM but maybe the new method must be limited to Java generators. @wing328 WDYT?

**Comment by borsch** [comment]
@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @nmuesch (2021/01) @wing328 please review

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java [hunk 1]
```diff
@@ -4198,7 +4212,7 @@ public CodegenParameter fromParameter(Parameter parameter, Set<String> imports)
             }
 
             // set default value
-            codegenParameter.defaultValue = toDefaultValue(parameterSchema);
+            codegenParameter.defaultValue = toDefaultParameterValue(parameterSchema);
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java:4215]: FYI. @OpenAPITools/generator-core-team

**Comment by wing328** [comment]
Please PM me via Slack when you've time. Thanks again for the PR, which mostly looks good.

**Review by wing328 — APPROVED** [review]
LGTM. As discussed I may file a PR later to consolidate the samples

**Comment by wing328** [comment]
@borsch thanks for the PR, which has been included in the v5.1.0 release: https://twitter.com/oas_generator/status/1373636654024380423

**Comment by wing328** [comment]
@Zomzog when you've time, can you please PM me via Slack? https://join.slack.com/t/openapi-generator/shared_invite/zt-12jxxd7p2-XUeQM~4pzsU9x~eGLQqX2g

**Pull request merged**